### PR TITLE
Core: define CREATURE_FAMILY_NONE to avoid magic numbers in code

### PR DESCRIPTION
**Pull request by ForesterDev**
**Changes proposed**:
- define CREATURE_FAMILY_NONE to avoid magic numbers in code

**Target branch(es)**: 335/6x

### CODE REVIEW

**Comment by Aokromes** [comment]
Broken compile.

File: src/server/scripts/Commands/cs_pet.cpp [hunk 1]
```diff
@@ -69,10 +69,10 @@ class pet_commandscript : public CommandScript
         }
 
         CreatureTemplate const* creatureTemplate = creatureTarget->GetCreatureTemplate();
-        // Creatures with family 0 crashes the server
-        if (!creatureTemplate->family)
+        // Creatures with family CREATURE_FAMILY_NONE crashes the server
+        if (creatureTemplate->family == CREATURE_FAMILY_NONE)
         {
-            handler->PSendSysMessage("This creature cannot be tamed. (family id: 0).");
+            handler->PSendSysMessage("This creature cannot be tamed. (family id: CREATURE_FAMILY_NONE).");
```
Comment (jackpoz) [inline src/server/scripts/Commands/cs_pet.cpp]: maybe it's useful to log both values in this case since the database doesn't store "CREATURE_FAMILY_NONE"

**Comment by Aokromes** [comment]
Any reason to don't merge this?

**Comment by jackpoz** [comment]
A git squash would have been nice...

**Pull request merged**